alonzo/impl/test/test/Test/Cardano/Ledger/Alonzo/Examples/Bbody.hs

```diff
@@ -0,0 +1,174 @@
+import Data.Default.Class (def)
+import qualified Data.Map.Strict as Map
+import qualified Data.Sequence.Strict as StrictSeq
+import Shelley.Spec.Ledger.API
+  ( BHBody (..),
+    BHeader (..),
+    Block (..),
+    DPState (..),
+    DState (..),
+    KESPeriod (..),
+    LedgerState (..),
+    Nonce (NeutralNonce),
+    OCert (..),
+    PrevHash (GenesisHash),
+    ProtVer (..),
+    UTxO (..),
+  )
+import Shelley.Spec.Ledger.BlockChain (bBodySize, mkSeed, seedEta, seedL)
+import Shelley.Spec.Ledger.EpochBoundary (BlocksMade (..))
+import Shelley.Spec.Ledger.Keys (KeyPair (..), KeyRole (..), coerceKeyRole, hashKey, signedDSIGN, signedKES)
+import Shelley.Spec.Ledger.LedgerState (UTxOState (..))
+import Shelley.Spec.Ledger.OCert (OCertSignable (..))
+import Shelley.Spec.Ledger.STS.Bbody (BbodyEnv (..), BbodyState (..))
+import Shelley.Spec.Ledger.Slot (BlockNo (..), SlotNo (..))
+import Shelley.Spec.Ledger.TxBody (TxIn (..))
+import Shelley.Spec.Ledger.UTxO (txid)
+import qualified Test.Cardano.Ledger.Alonzo.Examples.Utxow as UTXOW
+import Test.Shelley.Spec.Ledger.ConcreteCryptoTypes (C_Crypto)
+import Test.Shelley.Spec.Ledger.Generator.EraGen (genesisId)
+import Test.Shelley.Spec.Ledger.Utils
+  ( applySTSTest,
+    mkKESKeyPair,
+    mkKeyPair,
+    mkVRFKeyPair,
+    runShelleyBase,
+  )
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (Assertion, testCase, (@?=))
+
+type A = AlonzoEra C_Crypto
+
+-- =======================
+-- Setup the initial state
+-- =======================
+
+pp :: PParams A
+pp =
+  def
+    { _costmdls = Map.singleton PlutusV1 (CostModel mempty),
+      _maxValSize = 1000000000,
+      _maxTxExUnits = ExUnits 1000000 1000000,
+      _maxBlockExUnits = ExUnits 1000000 1000000
+    }
+
+bbodyEnv :: BbodyEnv A
+bbodyEnv = BbodyEnv pp def
+
+-- =======
+--  Tests
+-- =======
+
+dpstate :: DPState C_Crypto
+dpstate = def {_dstate = def {_rewards = Map.singleton UTXOW.scriptStakeCredSuceed (Coin 1000)}}
+
+initBBodyState :: BbodyState A
+initBBodyState = BbodyState (LedgerState UTXOW.initUtxoSt dpstate) (BlocksMade mempty)
+
+coldKeys :: KeyPair 'BlockIssuer C_Crypto
+coldKeys = KeyPair skCold vkCold
+  where
+    (vkCold, skCold) = mkKeyPair @C_Crypto (0, 0, 0, 0, 1)
+
+testBlock :: Block A
+testBlock = Block (BHeader bhb sig) txs
+  where
+    bhb =
+      BHBody
+        { bheaderBlockNo = BlockNo 0,
+          bheaderSlotNo = SlotNo 0,
+          bheaderPrev = GenesisHash,
+          bheaderVk = vKey coldKeys,
+          bheaderVrfVk = vvrf,
+          bheaderEta = coerce $ evalCertified () nonceNonce svrf,
+          bheaderL = coerce $ evalCertified () leaderNonce svrf,
+          bsize = fromIntegral $ bBodySize txs,
+          bhash = hashTxSeq txs,
+          bheaderOCert =
+            OCert
+              vkes
+              0
+              (KESPeriod 0)
+              (signedDSIGN @C_Crypto (sKey coldKeys) (OCertSignable vkes 0 (KESPeriod 0))),
+          bprotver = ProtVer 5 0
```

Comment:
Should this part, making the BHBody be a function that is reusable? Seems it is a function of txs.